src/client/pages/Project/Changelog/index.js

```diff
@@ -2,43 +2,52 @@
 
 import React, { Component } from 'react'
 import BranchSelect from 'components/BranchSelect'
-import ChangesetList from 'components/ChangesetList'
 import { connect } from 'react-redux'
 import Col from 'react-bootstrap/lib/Col'
 import Row from 'react-bootstrap/lib/Row'
 import Button from 'react-bootstrap/lib/Button'
-import ButtonGroup from 'react-bootstrap/lib/ButtonGroup'
-
+import { fetchChangelog } from 'ducks/repositories/actions'
+import { getChangelog } from 'ducks/repositories/selectors'
+import { fetchUsers } from 'ducks/users'
+import ChangesetList from 'components/ChangesetList'
+import LoadingComponent from 'components/LoadingComponent'
 
 export type Props = {
   params: Object,
-  data: Array<any>,
+  data: Array<any>
 };
 
 class Changelog extends Component {
   constructor(props: Props) {
     super(props)
 
     this.state = {
-      changesets: [],
       height: '100%',
       autoHideDuration: 4000,
       message: 'Hash has been copied to clipboard',
       open: false,
+      selectedChanges: 0,
+      shouldButtonUpdate: false,
     }
-    this.handleTouchTap = this.handleTouchTap.bind(this)
-    this.handleActionTouchTap = this.handleActionTouchTap.bind(this)
-    this.handleChangeDuration = this.handleChangeDuration.bind(this)
-    this.handleRequestClose = this.handleRequestClose.bind(this)
-    this.handleChange = this.handleChange.bind(this)
   }
 
-  props: Props
+  componentDidMount() {
+    if (this.props.params.splat) {
+      this
+        .props
+        .dispatch(fetchChangelog(this.props.params.splat))
+      this.props.dispatch(fetchUsers())
+    }
+  }
 
-  handleTouchTap() {
+  shouldComponentUpdate(nextProps, nextState) {
+    return this.state.selectedChanges !== nextState.selectedChanges
+  }
+
+  onSelectedChangesetsChange = (e) => {
     this.setState({
-      open: true,
-    })
+      selectedChanges: this.refs.changesetList.state.changesets.length,
```

Comment:
Avoid using `refs` unless you don't have any other choice. You could pass value with the event instead accessing component directly